scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -264,10 +264,18 @@ private void setPendingStatusesToAllEligibleTasks(List<EligibleTaskDescriptor> e
                                                                                            .equals(TaskStatus.PENDING))
                                                                       .collect(Collectors.toList());
 
-        notPendingYet.forEach(task -> task.getInternal().setStatus(TaskStatus.PENDING));
+        notPendingYet.forEach(task -> {
+            if (task.getInternal().getScheduledTime() == -1) {
+                task.getInternal().setScheduledTime(System.currentTimeMillis());
+            }
+            task.getInternal().setStatus(TaskStatus.PENDING);
+        });
 
         notPendingYet.forEach(task -> {
             getDBManager().updateTaskStatus(task, TaskStatus.PENDING);
+            getDBManager().updateScheduledTime(task.getJobId().longValue(),
+                                               task.getTaskId().longValue(),
+                                               task.getInternal().getScheduledTime());
```

Comment:
These will generate 2 transactions, making the operation non-atomic. I think it would be better to aggregate this in a single transaction if possible

Reply:
oooook